% Generated by roxygen2: do not edit by hand
% Please edit documentation in R/Cas_Type.R
\name{CaSilico}
\alias{CaSilico}
\title{In silico crRNA design for CRISPR-Cas12 (A, B and F1 subtypes) and CRISPR-Cas13 (A, B and D subtypes).}
\usage{
CaSilico(
  ResultsFolder = "CaSilico_output",
  TargetFasta = NULL,
  TargetAccession = NULL,
  TargetCoordinate = list(chromosome = NA, start = NA, end = NA, strand = NA, species =
    NA),
  CrisprTypes,
  ConservationMethod = 1,
  ConservationThreshold = 0.95,
  OffTarget = F,
  OffAsk = F,
  Organism = NULL,
  LocalOff = F,
  LocalFasta = c("Path1", "Path2"),
  LocalName = c("Organism1", "Organism2")
)
}
\arguments{
\item{ResultsFolder}{Name of the folder to save the final results; Default: ReaultsFolder="CaSilico_output".}

\item{TargetFasta}{Full path to the target sequence(s) to design crRNA for; Example: TargetFasta="/usr/mrb/casilico/genome.fasta"}

\item{TargetAccession}{Genome/gene/transcript accession numbers(s)(RefSeq or ENSEMBL ID)for the target sequence(s) to design crRNA for; Example: TargetAccession=c("U15717", "U15718").}

\item{TargetCoordinate}{Genomic coordinate(s)for the target sequence(s) to design crRNA for. CaSilico automatically retrieves the sequences based on the coordinates and their species; Example: TargetCoordinate=list(chromosome=c("1","5"), start=c("2000","12000"), end=c("2150","15500"), strand=c("-","-"), species = c("bos_taurus","bos_taurus")).}

\item{CrisprTypes}{A vector of CRSPR subtypes to design crRNAs based on their characteristics. One or more subtypes can be selected; Example: CrisprTypes=c("casVI_A","casVI_B","casVI_D","casV_A","casV_B","casV_F1").}

\item{ConservationMethod}{Desired method (1 or 2) to detect the conserved regions when more than one target sequence is submitted. See detailes for more information; Default: 1.}

\item{ConservationThreshold}{Desired threshold (between 0 and 1) to detect the conserved regions when more than one target sequence is submitted; Default: 0.95.}

\item{OffTarget}{If TRUE, CaSilico search for off-targets in the sequences specified by the Organism or LocalOff argument. This analysis may take a long time based on the number of subtypes in CrisprTypes argument and number of the selected sequences to be searched (Organism or LocalFasta arguments); Default: FALSE.}

\item{OffAsk}{If TRUE, in online mode before starting the off target analysis for each subtype, CaSilico ask from the user if he/she wants to continue or not; Default: FALSE.}

\item{Organism}{A vector of desired organisms for online off target analysis. CaSilico automatically search the identified crRNAs on the sequences(s)(NR or Refseq database, based on the CRISPR-Cas subtype) of the specified organismes; Example: Organism=c("Homo_sapiens","bos_tarus",...).}

\item{LocalOff}{If TRUE, CaSilico search for off targets in local mode and Organism argument dose not works. LocalFile have to be set. Default: FALSE.}

\item{LocalFasta}{Full path to the fasta file sequence(s) to be used for local off target analysis. One or more fasta file can be addressed; Example: LocalFasta=c("/yourpath1/filename1.fasta","/yourpath2/filename2.fasta").}

\item{LocalName}{A vector including desired names related to the LocalFasta argument to determine the name of the results. If it is'nt specified by the user, Casilico considers sequence1, sequence2 and etc based on the number of fasta files in LocalFasta argument; Example: LocalName=c("Name1", "Name2").}
}
\value{
Casilico presents the results in HTML and PDF files that can be found in ReaultsFolder.
}
\description{
In silico crRNA design for CRISPR-Cas12 (A, B and F1 subtypes) and CRISPR-Cas13 (A, B and D subtypes).
}
\section{References}{

...
}

\section{Further details}{

Casilico accepts three kinds of input formats as target sequence to be scanned for all potential target sites: 1) direct sequence(s) (TargetFasta argument); 2) Accession nember (TargetAccession argument) and 3) genomic coordinate (TargetCoordinate argument). One of these arguments have to be determined by the user as TargetFasta is recommended. Also, there are two different approaches to identify conserved regions (using ConservationMethod argument): 1) Most or all positions in the fragments (with length of w) have to be conserved and ≤2 positions can be polymorphic (based on the defined conservation score), as these positions can be occurred in the specific locations of the window, 2) Most or all positions in the fragments (with length of w) have to be conserved and ≤2 positions can be polymorphic (based on the defined conservation score) in everywhere of the fragment (with length of w).
}

\examples{

# Run with TargetFasta

data <- paste0(system.file(package = "CaSilico"),"/data/3D.fasta")

CaSilico(ResultsFolder="Example",
         TargetFasta=data,
         TargetAccession=NULL,
         CrisprTypes=c("casVI_A"),
         ConservationMethod = 1,
         ConservationThreshold=0.98,
         OffTarget = F,
         OffAsk = F)




# Run with TargetFasta and off-target analysis

data <- paste0(system.file(package = "CaSilico"),"/data/3D.fasta")

genome_dir <- paste0(system.file(package = "CaSilico"),"/data/genomes_off_target")

Local_Fasta <- file.path(genome_dir,list.files(genome_dir))

Local_Name <- c("Genome1","Genome2","Genome3")

CaSilico(ResultsFolder="Example2",
         TargetFasta=data,
         TargetAccession=NULL,
         CrisprTypes=c("casVI_A","casVI_B","casVI_D","casV_A","casV_B","casV_F1"),
         ConservationMethod = 1,
         ConservationThreshold=0.98,
         OffTarget = T,
         OffAsk = F,
         Organism=NULL,
         LocalOff=T,
         LocalFasta=Local_Fasta,
         LocalName=Local_Name)




 # Run with accession numbers

 CaSilico(ResultsFolder="Example3",
          TargetAccession=c("U15717", "U15718"),
          CrisprTypes=c("casVI_A","casVI_B"),
          ConservationMethod = 1,
          ConservationThreshold=0.98,
          OffTarget = F,
          OffAsk = F)




 # Run with sequence coordinate
 CaSilico(ResultsFolder="Example4",
          TargetCoordinate=list(chromosome=c("1","1"),
                             start=c("2000","2000"),
                             end=c("2150","2150"),
                             strand=c("-","-"),
                             species = c("bos_taurus","bos_taurus")),
          CrisprTypes=c("casVI_B","casVI_A"),
          ConservationMethod = 1,
          ConservationThreshold=0.98,
          OffTarget = F,
          OffAsk = F)


}
